### ISSUE TITLE
PC app features

### ISSUE THREAD

**Issue by nastys** [roxas75/rxTools#173]
So... there is going to be a (actually two) companion app, like @AlbertoSONIC suggested, but it will be optional, so the user could put the files manually if they want, like @173210 suggested.
Anyway, the app would be almost useless if it could only download the files for MSET and such, so I think we should make it more interesting.
I was thinking that the app should take care of the initial setup, manage the installed themes (BTW, it could load a PNG file from the selected theme folder :smile: ) as well as download new ones (it would need a server, though, or perhaps a git repository?), change the configuration, manage the emuNAND...

That would be a worthwhile PC app :smirk:

**Comment by AlbertoSONIC** [comment]
I agree everything but the "Compile" option. If an user is able to set up the environment, he's able to compile it himself. I think it's better to rely on a build server, which automatically builds at every change and hosts the output of the compilation. And here I call @filfat :)

**Comment by AlbertoSONIC** [comment]
Ehm... Just found out that his build server is offline since he left the 3ds scene... Then we have to find someone to host a build server

**Comment by nastys** [comment]
@AlbertoSONIC I agree on the build server but in my opinion there's nothing wrong with a "Compile" option. If a user want the latest version and it's not been built yet or if the server is down, they'll only have to press that button.

**Comment by AlbertoSONIC** [comment]
They'd also have to install Imagemagick, devkitpro, mingw...

**Comment by nastys** [comment]
If anything's missing, the button will not appear.

**Comment by AlbertoSONIC** [comment]
:+1:

**Comment by capito27** [comment]
How about adding also a detection system to check what compiling
requirements are present (check if image magic is present, devkitpro,mingw
for Windows as well) and if not give a link

(Check and edit appropriately the path string as well?)

**Comment by 173210** [comment]
@nastys Exactly, but it doesn't make sense to add an option nobody uses. At least I won't. It just makes difficult to change the building system. And you shouldn't have time for that enough to make a better terminal emulator than GNOME terminal or something which already exists.

**Comment by 173210** [comment]
By the way, if users compiles rxTools themselves, it will be because users want to try the latest version. Why don't you provide nightly builds with Travis CI and rxtools.net?

**Comment by nastys** [comment]
@173210 Which option are you talking about?
I'll provide both ways of getting the latest version of rxTools. Building will only be available if the devkit ha been set up correctly.

**Comment by 173210** [comment]
@nastys compiling.

**Comment by nastys** [comment]
What do you mean by "It just makes difficult to change the building system."?
It will just:
- clone the latest repo
- run "make release"

Even easy to code...

**Comment by 173210** [comment]
@nastys It's a problem. I rarely run the command. I'm often use `make <TARGET I REALLY WANT>`. GUI is just worse for me.

**Comment by nastys** [comment]
@173210 Then just don't use it...
This isn't really for developing rxTools, it's for testing it.

**Comment by 173210** [comment]
@nastys Then nightly builds should be enough. Travis CI can provide a build for each commits.

**Comment by nastys** [comment]
I never said they're not, but it's still useful in case the server are down.

**Comment by 173210** [comment]
@nastys Is it worth to implement only for the case?

**Comment by filfat** [comment]
@AlbertoSONIC Oh, you ping'd me that explains the thousands of emails :P 
Anyhow yes the server is down I might reconfigure it if I ever get back to the homebrew community but that isn't really an option now (need all the resources I can get for work).

Back to a swiftly tailored playlist, eh... hehehe.

**Comment by nastys** [comment]
@173210 It's not difficult to do, really...

**Comment by AlbertoSONIC** [comment]
@filfat :+1: :D

**Comment by 173210** [comment]
@nastys I doubt that. Users should set up development environment by themselves, so you may have to provide a tutorial for them.
UI implementation and interface with another part of rxTools would be necessary. Moreover, the feature will highly depend on the environment.

And it's not all. You must maintain all of them. It's not so simple just to run commands.

**Comment by AlbertoSONIC** [comment]
Anyhow I'd just go for the build server only. As I said, if an user is able to setup the environment, he's also able to run "make clean && make release"

**Comment by nastys** [comment]
Fine, I'll implement this later (maybe...)

**Comment by Syphurith** [comment]
You could use "official nightly builds" or something else that be traggered when github updates.
I personally don't like users to compile the source code.. However Theme related operations could be done in such an app, thus may even get someone away from imagemagick.
And it could grab firm/mset or whatever from other sites.. or even tutorials could be updated..
You could decide how to implement an official PC App, and let it be in another repo.
